src/box/txn.c

```diff
@@ -1601,3 +1616,11 @@ txn_attach(struct txn *txn)
 	trigger_add(&fiber()->on_yield, &txn->fiber_on_yield);
 	trigger_add(&fiber()->on_stop, &txn->fiber_on_stop);
 }
+
+void
+txn_stmt_mark_as_temporary(struct txn *txn, struct txn_stmt *stmt)
+{
+	/* Revert row counter increases. */
+	txn_update_row_counts(txn, stmt, TXN_ROW_COUNT_DEC);
```

Comment:
Please, add an assert, that `stmt->row` is not `NULL` and `is_temporary` is `false` now. To ensure it was accounted before you unaccount it.

Reply:
done